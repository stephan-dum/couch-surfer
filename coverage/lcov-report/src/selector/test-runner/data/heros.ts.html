<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/selector/test-runner/data/heros.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">src/selector/test-runner/data</a> heros.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export default [
  { "name": "mario", "_id": "mario", "rank": 5, "series": "mario", "debut": 1981 },
  { "name": "jigglypuff", "_id": "puff", "rank": 8, "debut": 1996, "retired": true },
  { "name": "link", "rank": 10, "_id": "link", "series": "zelda", "debut": 1986 },
  { "name": "donkey kong", "rank": 7, "_id": "dk", "series": "mario", "debut": 1981 },
  { "name": "pikachu", "series": "pokemon", "_id": "pikachu", "rank": 1, "debut": 1996 },
  { "name": "captain falcon", "_id": "falcon", "rank": 4, "series": "f-zero", "debut": 1990 },
  { "name": "luigi", "rank": 11, "_id": "luigi", "series": "mario", "debut": 1983 },
  { "name": "fox", "_id": "fox", "rank": 3, "series": "star fox", "debut": 1993 },
  { "name": "ness", "rank": 9, "_id": "ness", "series": "earthbound", "debut": 1994 },
  { "name": "samus", "rank": 12, "_id": "samus", "series": "metroid", "debut": 1986 },
  { "name": "yoshi", "_id": "yoshi", "rank": 6, "series": "mario", "debut": 1990 },
  { "name": "kirby", "_id": "kirby", "series": "kirby", "rank": 2, "debut": 1992 }
];
&nbsp;
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 29 2019 20:49:20 GMT+0100 (GMT+01:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
<script src="..\..\..\..\block-navigation.js"></script>
</body>
</html>
