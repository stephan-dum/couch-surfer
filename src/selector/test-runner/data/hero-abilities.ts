

interface HeroAbility {
  hero: string;
  eats?: string[];
  wears?: IClothes[];
  strength: number;
  speed: number;
  results: number[];
}

interface IClothes {
  name: string;
  defense?: number;
  attack?: number;
  quality?: number;
}

export const peach = {
  hero: "peach",
  wears : [
    {
      name : "dress",
      defense : 1,
      quality : 100
    },
    {
      name : "umbrella",
      attack : 1,
      defense : 4
    }
  ],
  eats : ['cake', 'turnips', 'sweets'],
  strength : 5,
  speed: 10,
  results : [ 75, 82, 87 ],
};

export const mario = {
  hero : "mario",
  wears : [
    {
      name : "red shirt",
      defense: 2,
    },
    {
      name : "blue pants",
      defense: 2,
    }
  ],
  eats : ["coins", "mushrooms"],
  strength : 15,
  speed: 10,
  results: [ 82, 85, 88 ],
};

export const kerby = {
  hero: "kirby",
  eats : ["everything"],
  strength : 10,
  speed: 7,
  results : [ 82, 86, 10 ],
};

const heroAbilities: HeroAbility[] = [
  peach,
  mario,
  kerby,
];

export default heroAbilities;