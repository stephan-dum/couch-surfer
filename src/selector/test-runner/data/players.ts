export default [
  { name: 'James', _id: 'james',  favorites: ['Mario', 'Pokemon'], age: 20 },
  { name: 'Mary', _id: 'mary',  favorites: ['Pokemon'], age: 21 },
  { name: 'Link', _id: 'link', favorites: ['Zelda', 'Pokemon'], age: 22 },
  { name: 'William', _id: 'william', favorites: ['Mario'], age: 23 }
];