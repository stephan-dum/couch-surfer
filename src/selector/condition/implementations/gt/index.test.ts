import { ISelectorTestSuite } from "~/selector/test-runner/index.d";
import executeSuite from "~/selector/test-runner";
import data from "~/selector/test-runner/data/players";

const valid = [
  { name: 'William', _id: 'william', favorites: ['Mario'], age: 23}
];

const defaultTestSuit = {
  args : ["age"],
  execute : [{
    args: [22],
    data,
    expected: valid,
  }]
};

const testSuiteCollection: ISelectorTestSuite[] = [
  {
    title : "gt",
    selector : {
      age: {
        "$gt" : "$age"
      }
    },
    ...defaultTestSuit,
  },
];

describe("gt", () => {
  executeSuite(testSuiteCollection)
});

export default testSuiteCollection;