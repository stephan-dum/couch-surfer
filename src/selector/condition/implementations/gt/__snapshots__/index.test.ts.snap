// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"doc => {
  return doc.age > age;
};"
`;
